### DISCUSSION TITLE
How to get ALL follows for an actor?

### DISCUSSION THREAD

**Discussion by lassehjorthmadsen** [bluesky-social/atproto#2012] [How to do it?] [answered]
It [looks](https://github.com/bluesky-social/atproto/blob/main/lexicons/app/bsky/graph/getFollows.json) like you can only get 100 follows for a given actor. I really need to get ALL follows. I would imagine that you could do a series of requests with parameters like first_follows=101, last_follows=200, but that doesn't seem to be the case. Any way around this?
Thanks,
Lasse

**Comment by dholms** [comment, marked as answer]
It's a paginated API so you can enumerate all followers by using the `cursor` from the previous response as a param in the next request. Once the API returns an undefined cursor, pagination is finished 👍

**Reply by GiuseppeP96** [reply]
i'm using the cursor but the list is not really complete. I can only get 637 over the 672  on the follower count on my profile page.

**Reply by DavidBuchanan314** [reply]
I believe the count on the profile also includes blocked accounts, and *maybe* also suspended or deleted accounts (less sure about that), whereas getFollows will not.

**Reply by GiuseppeP96** [reply]
I don't think this is the case. I've randomly checked from follows API for non-mutuals and as a result i get some users which are still following me (and vice versa) but not returned by get_followers API.

**Reply by DavidBuchanan314** [reply]
can you give an example?

**Reply by dabartmess** [reply]
@DavidBuchanan314  There's a (if you use Python, other languages it'll be different names) a dict item in the followers schema called ["viewers"]["blocked_by"] OR ["blocking"] which can help differentiate those accounts...

**Comment by Skatterbrainz** [comment]
Is there an example for using the cursor to effectively run paging requests?
